src/components/button.component.js

```diff
@@ -0,0 +1,218 @@
+    },
+  },
+  danger: {
+    buttonStyle: {
+      borderColor: '#dc3545',
+      borderWidth: 1,
+      backgroundColor: '#c9302c',
+    },
+    disabledStyle: {
+      borderColor: '#d43f3a',
+      borderWidth: 1,
+      backgroundColor: '#d9534f',
+    },
+  },
+  warning: {
+    buttonStyle: {
+      borderColor: '#eea236',
+      borderWidth: 1,
+      backgroundColor: '#ec971f',
+    },
+    disabledStyle: {
+      borderColor: '#eea236',
+      borderWidth: 1,
+      backgroundColor: '#f0ad4e',
+    },
+  },
+  info: {
+    buttonStyle: {
+      borderColor: '#46b8da',
+      borderWidth: 1,
+      backgroundColor: '#31b0d5',
+    },
+    disabledStyle: {
+      borderColor: '#46b8da',
+      borderWidth: 1,
+      backgroundColor: '#5bc0de',
+    },
+  },
+  success: {
+    buttonStyle: {
+      borderColor: '#4cae4c',
+      borderWidth: 1,
+      backgroundColor: '#449d44',
+    },
+    disabledStyle: {
+      borderColor: '#4cae4c',
+      borderWidth: 1,
+      backgroundColor: '#5cb85c',
+    },
+  },
+  primary: {
+    buttonStyle: {
+      borderColor: '#2e6da4',
+      borderWidth: 1,
+      backgroundColor: '#286090',
+    },
+    disabledStyle: {
+      borderColor: '#2e6da4',
+      backgroundColor: '#337ab7',
+    },
+  },
+};
+
+const sizes = {
+  small: {
+    buttonStyle: {
+      paddingTop: 5,
+      paddingBottom: 5,
+      paddingEnd: 8,
+      paddingLeft: 8,
+      marginLeft: 4,
+      marginRight: 4,
+    },
+    textStyle: {
+      fontSize: normalize(12),
+    },
+    iconStyle: {
+      size: 12,
+    },
+  },
+  default: {
+    buttonStyle: {
+      paddingTop: 8,
+      paddingBottom: 8,
+      paddingRight: 14,
+      paddingLeft: 14,
+      marginLeft: 6,
+      marginRight: 6,
+    },
+    textStyle: {
+      fontSize: normalize(14),
+    },
+    iconStyle: {
+      size: 16,
+    },
+  },
+  large: {
+    buttonStyle: {
+      paddingTop: 13,
+      paddingBottom: 13,
+      paddingRight: 17,
+      paddingLeft: 17,
+      marginLeft: 8,
+      marginRight: 8,
+    },
+    textStyle: {
+      fontSize: normalize(16),
+    },
+    iconStyle: {
+      size: 20,
+    },
+  },
+};
+
+const defaultButtonStyle = {
+  borderWidth: 1,
+  borderRadius: 2,
+};
+
+const defaultTextStyle = Platform.select({
+  android: {
+    ...fonts.fontPrimarySemiBold,
+  },
+  ios: {
+    ...fonts.fontPrimarySemiBold,
+  },
+  color: colors.white,
+});
+
+const defaultIconStyle = {
+  style: {
+    marginRight: 4,
+  },
+};
+
+const defaultDisabledStyle = {
+  borderWidth: 1,
+};
+
+const defaultDisabledTextStyle = {
+  color: '#d2d2d2',
+};
+
+export class Button extends Component {
+  props: {
+    icon: string,
+    size: string,
+    type: string,
+    title: string,
+    disabled: boolean,
+  };
+
+  render() {
+    const size = this.props.size;
+    const type = this.props.type;
+    const icon = this.props.icon;
+    const title = this.props.title;
+
+    const isAndroid = Platform.OS === 'android';
```

Comment:
If we're using `'android'` anywhere else we should move into a constants file.

Reply:
We're using 'android' and 'ios' in several parts, but I don't feel like we need constants for these.. 🤔